egs/wsj/s5/steps/nnet3/xconfig_to_configs.py

```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/env python
+
+# we're using python 3.x style print but want it to work in python 2.x,
+from __future__ import print_function
+import os
+import argparse
+import shlex
+import sys
+import warnings
+import copy
+import imp
+import ast
+from collections import defaultdict
+
+sys.path.insert(0, 'steps/nnet3/libs/')
+# the following is in case we weren't running this from the normal directory.
+sys.path.insert(0, os.path.realpath(os.path.dirname(sys.argv[0])) + '/libs/')
+
+import xconfig_utils
+import xconfig_layers
+
+
+def GetArgs():
+    # we add compulsary arguments as named arguments for readability
+    parser = argparse.ArgumentParser(description='Reads an xconfig file and creates config files '
+                                     'for neural net creation and training',
+                                     epilog='Search egs/*/*/local/nnet3/*sh for examples')
+    parser.add_argument('xconfig_file',
+                        help='Filename of input xconfig file')
+    parser.add_argument('config_dir',
+                        help='Directory to write config files and variables')
+
+    print(' '.join(sys.argv))
+
+    args = parser.parse_args()
+    args = CheckArgs(args)
+
+    return args
+
+def CheckArgs(args):
+    if not os.path.exists(args.config_dir):
+        os.makedirs(args.config_dir)
+    return args
+
+
+#     # write the files used by other scripts like steps/nnet3/get_egs.sh
+#     f = open(config_dir + 'vars', 'w')
+#     print('model_left_context=' + str(left_context), file=f)
+#     print('model_right_context=' + str(right_context), file=f)
+#     print('num_hidden_layers=' + str(num_hidden_layers), file=f)
```

Comment:
if we are not supporting layer-wise pretraining this would be an unnecessary variable. Also it does not have a proper interpretation in a non-linear network.